uportal-war/src/main/java/org/jasig/portal/security/provider/AnyUnblockedGrantPermissionPolicy.java

```diff
@@ -373,86 +367,36 @@ private boolean containsType(final Set<IPermission> permissions, final String so
-            this.principalName = principalName;
-            this.owner = owner;
-            this.activity = activity;
-            this.target = target;
-        }
-        @SuppressWarnings("unused")
-        public String getPrincipalName() {
-            return principalName;
-        }
-        @SuppressWarnings("unused")
-        public String getOwner() {
-            return owner;
-        }
-        @SuppressWarnings("unused")
-        public String getActivity() {
-            return activity;
-        }
-        @SuppressWarnings("unused")
-        public String getTarget() {
-            return target;
-        }
-        @Override
-        public int hashCode() {
-            final int prime = 31;
-            int result = 1;
-            result = prime * result
-                    + ((activity == null) ? 0 : activity.hashCode());
-            result = prime * result + ((owner == null) ? 0 : owner.hashCode());
-            result = prime * result
-                    + ((principalName == null) ? 0 : principalName.hashCode());
-            result = prime * result
-                    + ((target == null) ? 0 : target.hashCode());
-            return result;
-        }
-        @Override
-        public boolean equals(Object obj) {
-            if (this == obj)
-                return true;
-            if (obj == null)
-                return false;
-            if (getClass() != obj.getClass())
-                return false;
-            CacheTuple other = (CacheTuple) obj;
-            if (activity == null) {
-                if (other.activity != null)
-                    return false;
-            } else if (!activity.equals(other.activity))
-                return false;
-            if (owner == null) {
-                if (other.owner != null)
-                    return false;
-            } else if (!owner.equals(other.owner))
-                return false;
-            if (principalName == null) {
-                if (other.principalName != null)
-                    return false;
-            } else if (!principalName.equals(other.principalName))
-                return false;
-            if (target == null) {
-                if (other.target != null)
-                    return false;
-            } else if (!target.equals(other.target))
-                return false;
-            return true;
-        }
-        @Override
-        public String toString() {
-            return "CacheTuple [principalName=" + principalName + ", owner=" + owner + ", activity=" + activity
-                    + ", target=" + target + "]";
+    protected CacheKey getCacheKey(IAuthorizationPrincipal principal, IPermissionOwner owner,
+                                   IPermissionActivity activity, IPermissionTarget target) {
+
+        CacheKey rslt;
+        if (principal.isGroup()) {
+            /*
+             * Untagged keys for groups...
+             */
+            rslt = CacheKey.build(
```

Comment:
@ChristianMurphy -- The page you linked is badly in need of more explanation.  It's clear what he's suggesting, but the rationale is entirely missing.  (Perhaps he just imagines that once you see his brilliance, no explanation is required?... I don't buy it.)

In the history of Java programming, there have been many luminaries (through the years) who have staunchly advocated a style where methods should have at most one `return` statement (also for the purpose of "clarity").  That advice is starkly contradictory to the advice you linked.  You really can follow only one or the other.

Now then...
  - I don't feel more obligation to follow the "one return per method" folks than the advice you posted;  I didn't bring it up because I'm a "one return per method" zealot, only to show that what you posted is not universally accepted.
  - I don't feel obligated to follow _anyone's_ Java style advice, simply because s/he said it;  I _deeply enjoy_ opportunities for growth and a programmer... but if you want to persuade me of something, you have to make a compelling case.

This link does not make a compelling case.  Do you perhaps have another?  I'm open minded... but also always critical.